apps/performance-tests/src/hierarchies/StatelessHierarchyProvider.ts

```diff
@@ -16,17 +15,30 @@ import {
   HierarchyNode,
   HierarchyProvider,
 } from "@itwin/presentation-hierarchies";
-import { createCachingECClassHierarchyInspector, ECClassHierarchyInspector, ECSchemaProvider } from "@itwin/presentation-shared";
+import {
+  createCachingECClassHierarchyInspector,
+  EC,
+  ECClassHierarchyInspector,
+  ECSchemaProvider,
+  ECSqlQueryDef,
+  ECSqlQueryExecutor,
+  ECSqlQueryReaderOptions,
+} from "@itwin/presentation-shared";
+import { asyncScheduler, expand, filter, finalize, from, observeOn, of, tap } from "rxjs";
 import { LOGGER } from "../util/Logging";
 
-export interface ProviderOptions {
-  iModel: IModelDb;
+export interface ProviderOptionsBase {
   rowLimit?: number | "unbounded";
   getHierarchyFactory(imodelAccess: ECSchemaProvider & ECClassHierarchyInspector): HierarchyDefinition;
   filtering?: {
     paths: HierarchyFilteringPath[];
   };
 }
+export type ProviderOptionsWithIModel = { iModel: IModelDb } & ProviderOptionsBase;
+
+export type ProviderOptionsWithIModelAccess = { imodelAccess: IModelAccess } & ProviderOptionsBase;
+
+export type ProviderOptions = ProviderOptionsWithIModel | ProviderOptionsWithIModelAccess;
```

Comment:
do we even need these to be exported? from the first look it seems they should only be needed here (the `setup` functions probably don't need to specify the return type).

Reply:
If return type of `setup` is not specified, then vs code complains:
![image](https://github.com/user-attachments/assets/b5d81efc-b948-468c-a3e1-7e67eec5dae3)

Reply:
It seems simpler to just export those types and specify the return type of `setup`, or do you still think we should not export them?

Reply:
Short term it may be simpler, but we're littering intellisense with a number of useless types, which will make it more complicated in the future.

The build problems can be fixed:
- remove `rowLimit` from the object returned by `setup`,
- define types for `getHierarchyFactory`:
  ```ts
  getHierarchyFactory: (imodelAccess: ECSchemaProvider & ECClassHierarchyInspector): HierarchyDefinition => ({ ... 
  ```
- supply `rowLimit` when creating `StatelessHierarchyProvider`:
  ```ts
  const provider = new StatelessHierarchyProvider({
    ...props,
    rowLimit: "unbounded",
  });
  ```

Reply:
Applied suggestion